src/main/scala/io/iohk/ethereum/consensus/ConsensusAdapter.scala

```diff
@@ -0,0 +1,69 @@
+package io.iohk.ethereum.consensus
+
+import monix.eval.Task
+import monix.execution.Scheduler
+
+import io.iohk.ethereum.blockchain.sync.regular.BlockImportFailed
+import io.iohk.ethereum.blockchain.sync.regular.BlockImportResult
+import io.iohk.ethereum.blockchain.sync.regular.DuplicateBlock
+import io.iohk.ethereum.domain.Block
+import io.iohk.ethereum.domain.BlockHeader
+import io.iohk.ethereum.domain.BlockchainReader
+import io.iohk.ethereum.ledger.BlockExecutionError.ValidationBeforeExecError
+import io.iohk.ethereum.ledger.BlockExecutionSuccess
+import io.iohk.ethereum.ledger.BlockQueue
+import io.iohk.ethereum.ledger.BlockValidation
+import io.iohk.ethereum.utils.BlockchainConfig
+import io.iohk.ethereum.utils.FunctorOps._
+import io.iohk.ethereum.utils.Logger
+
+/** This is a temporary class to isolate the real Consensus and extract responsibilities which should not
+  * be part of the consensus in the final design, but are currently needed.
+  */
+class ConsensusAdapter(
+    consensus: Consensus,
+    blockchainReader: BlockchainReader,
+    blockQueue: BlockQueue,
+    blockValidation: BlockValidation,
+    validationScheduler: Scheduler
+) extends Logger {
+  def evaluateBranchBlock(
+      block: Block
+  )(implicit blockExecutionScheduler: Scheduler, blockchainConfig: BlockchainConfig): Task[BlockImportResult] =
+    blockchainReader.getBestBlock() match {
+      case Some(bestBlock) =>
+        if (isBlockADuplicate(block.header, bestBlock.header.number)) {
+          log.debug("Ignoring duplicated block: {}", block.idTag)
+          Task.now(DuplicateBlock)
+        } else {
+          doBlockPreValidation(block).flatMap {
+            case Left(error) =>
+              Task.now(BlockImportFailed(error.reason.toString))
+            case Right(BlockExecutionSuccess) =>
+              consensus.evaluateBranch(Seq(block))
+          }
+        }
+      case None =>
+        log.error("Couldn't find the current best block")
+        Task.now(BlockImportFailed("Couldn't find the current best block"))
+    }
+
+  private def doBlockPreValidation(block: Block)(implicit
+      blockchainConfig: BlockchainConfig
+  ): Task[Either[ValidationBeforeExecError, BlockExecutionSuccess]] =
+    Task
+      .evalOnce(blockValidation.validateBlockBeforeExecution(block))
+      .tap {
+        case Left(error) =>
+          log.error("Error while validating block with hash {} before execution: {}", block.hash, error.reason)
```

Comment:
error here is of type ValidationBeforeExecError right? then it should be `error.reason.toString`

Reply:
But `toString` will be called by the logger, right ?

Reply:
I added the toString and also wrote block hashes in hex instead of plain ByteString